Geometry/MTDCommonData/src/MTDBaseNumber.cc

```diff
@@ -0,0 +1,54 @@
+#include "Geometry/MTDCommonData/interface/MTDBaseNumber.h"
+
+MTDBaseNumber::MTDBaseNumber():_theLevels(0) { } 
+
+MTDBaseNumber::MTDBaseNumber(const MTDBaseNumber & aBaseNumber):
+  _sortedName(aBaseNumber._sortedName),
+  _sortedCopyNumber(aBaseNumber._sortedCopyNumber),  
+  _theLevels(aBaseNumber._theLevels) { }  
+
+void MTDBaseNumber::setSize(const int & size) {
+  _sortedName.resize(size);
+  _sortedCopyNumber.resize(size);
+}
+
+void MTDBaseNumber::addLevel(const std::string& name, const int & copyNumber)
+{
+  _sortedName[_theLevels] = name;
```

Comment:
@ianna again this is essentially a clone of EcalBaseNumber, and as far as I can see MuonBaseNumber is not that different. Setting a priori the size means coupling the geometry structure with the code. One could put some reasonable maxLevel as limit to allow the insertion, what do you think?

Concerning the struct as this is already a class I am not sure which could be the benefit, could you please clarify in case?